backend/substrapp/tasks/tasks.py

```diff
@@ -633,12 +615,12 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
     pred_path = path.join(subtuple_directory, 'pred')
     opener_file = path.join(subtuple_directory, 'opener/opener.py')
     algo_path = path.join(subtuple_directory)
-    algo_docker = f'substra/algo_{algo_hash[0:8]}'.lower()  # tag must be lowercase for docker
+    algo_docker = get_algo_image_name(algo_hash)  # tag must be lowercase for docker
     algo_docker_name = f'{tuple_type}_{subtuple["key"][0:8]}'
     output_head_model_filename = 'head_model'
     output_trunk_model_filename = 'trunk_model'
 
-    remove_image = not((compute_plan_id is not None and rank != -1) or settings.TASK['CACHE_DOCKER_IMAGES'])
+    remove_image = not(compute_plan_id is not None or settings.TASK['CACHE_DOCKER_IMAGES'])
```

Comment:
Not directly link with this PR but this `remove_image` value is now only use one time so we might want to set it either directly when calling `compute_docker` or just above it.

Reply:
Done